content/200-concepts/100-components/02-prisma-client/260-client-extensions/120-query.mdx

```diff
@@ -147,6 +147,30 @@ const xprisma = prisma.$extends({
 })
 ```
 
+### Mutate the result of a query
+
+You can use `await` and then mutate the result of the `query` promise, because its value is type-safe. TypeScript catches any unsafe mutations on the object.
```

Comment:
@millsp  Should we add a caveat at the start of this line, like "If you use TypeScript, then [...]"?

Reply:
I think it would be fine to just remove it? I think we do emphasize about the type-safety elsewhere, do we ?
```suggestion
You can use `await` and then mutate the result of the `query` promise.
```